pkg/controller/proposal/controller.go

```diff
@@ -480,6 +480,8 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
 			for path, changeValue := range changeValues {
 				config.Values[path] = changeValue
 			}
+			config.Values = tree.PrunePathMap(config.Values, true)
```

Comment:
I think you can operate on change values first before putting it in the config values?

Reply:
I'm not sure I can. The change values could contain just the delta no? We want the prune the entire config as a whole.

Reply:
that is right,